[IMP] accounting: update online payment and restructure payment (#17134)
odoo/documentation: content/applications/finance/accounting/payments.rst

===============

Several payment methods are available in Odoo to allow different configurations for different types
of payments. Payment methods include manual payments (such as cash),

> :doc:`checks <payments/pay_checks>`, and batch payment files (such as :ref:`NACHA

Reviewer: ```suggestion
of payments. Payment methods include manual payments (such as cash), :doc:`checks 
<payments/pay_checks>`, and batch payment files (such as :ref:`NACHA
```

Preferred payment method
------------------------

A preferred payment method can be defined on a contact so that it is automatically selected when a

> payment is created for that contact. From the search bar, invoices and bills can be grouped

Reviewer: ```suggestion
payment is created for that contact. From the search bar, invoices and bills can be grouped by
```

   - :doc:`Batch payments <payments/batch>` can simplify reconciling deposits containing multiple
     checks.

.. _accounting/payments/preferred-payment-methods:

Preferred payment method

> ------------------------

Reviewer: I think it was useful to have this section before the Checks section, because this one is more general and applies to all payment options, and checks are more specific.

A preferred payment method can be defined on a contact so that it is automatically selected when a
payment is created for that contact. Invoices and bills can be filtered by
:guilabel:`Preferred Payment Method Line` to simplify

> :ref:`group <accounting/payments/group-payments>` payments.

Reviewer: I don't see this filter (Preferred 
Payment Method Line) in the Invoices/Bills list view:

<img width="687" height="396" alt="Image" src="https://github.com/user-attachments/assets/e28e2ff9-09f9-4392-9a78-4e3f0007d3d5" />

```suggestion
payment is created for that contact. Invoices and bills can be filtered by :guilabel:`Payment
Method` to simplify :ref:`group <accounting/payments/group-payments>` payments.
```

      .. tip::
         If the main bank account is set as the outstanding account on the bank journal's payment
         method, registering the full payment on an invoice or bill moves the invoice/bill directly
         to the :guilabel:`Paid` status without requiring bank reconciliation.

      .. note::

         - Unreconciling a payment unlinks it from the invoice or bill but does not delete the
           payment.
         - If a payment is (un)reconciled in a different currency, a journal entry is automatically
           created to post the currency exchange gains/losses (reversal) amount.
         - If a payment is (un)reconciled on an invoice with cash-basis taxes, a journal entry is
           automatically created to post the cash-basis tax (reversal) amount.

.. _accounting/payments/online:

Online payments
---------------

To make invoice payment more convenient, the :guilabel:`Invoice Online Payment` feature can be
enabled, adding a :guilabel:`Pay Now` button to the customer portal. This allows customers to view
invoices online and pay directly using their preferred payment method, simplifying the payment
process.

> Make sure :ref:`payment providers <payment_providers/supported_providers>` are configured correctly.

Reviewer: I think I would make a note of it. What do you think?
If so, you can add it to the note below (as 1st bullet and the Wire transfer information as 2nd)

scroll down to the :guilabel:`Customer Payments` section to enable
:guilabel:`Invoice Online Payment`, and click :guilabel:`Save`.

After creating the invoice, click :guilabel:`Send` to email it to the customer, who receives an
email with a link to view the invoice on their customer portal. From there, they can click
:guilabel:`Pay Now` to select a :ref:`Payment Provider <payment_providers/online_providers>`.

.. note::
   By default, :ref:`Wire Transfer <payment_providers/bank_payments/wire_transfer>` is the only
   enabled payment provider, but payment details must still be entered.

.. seealso::
   :doc:`/applications/finance/payment_providers`

As soon as the customer has paid, a payment for the invoice is registered, and the invoice is marked
as :guilabel:`In payment`.

.. tabs::

   .. group-tab:: Without outstanding accounts

      If the payment method configured in the payment provider has no :ref:`outstanding accounts
      <accounting/journals/outstanding-accounts>`:

      - No journal entry created.
      - The :guilabel:`Amount Due` is not updated.
      - After :ref:`reconciliation <accounting/payments/reconciling-payments>` between the payment

>         and the bank transaction, the invoice is marked as :guilabel:`Paid`.

Reviewer: In this case, the reconciliation is between the invoice and the bank as there are no outstanding accounts.
```suggestion
      - After the invoice is :doc:`reconciled <bank/reconciliation>` with a bank transaction, the 
        invoice is marked as :guilabel:`Paid`.
```

   .. group-tab:: Without outstanding accounts

      If no :ref:`outstanding accounts <accounting/journals/outstanding-accounts>` are configured:

      - No journal entry is created.
      - The :guilabel:`Amount Due` is not updated.

      More payment information is accessible by clicking the :guilabel:`Payments` smart button.

   .. group-tab:: Using outstanding accounts

      If :ref:`outstanding accounts <accounting/journals/outstanding-accounts>` are configured:

      - A journal entry is created.
      - The counterpart is reflected in an
        :ref:`outstanding receipts or payments account <accounting/journals/outstanding-accounts>`.
      - The :guilabel:`Amount Due` is reduced based on the payment amount.

      More payment information is also accessible by clicking the :icon:`fa-info-circle`
      :guilabel:`(information icon)` and clicking :guilabel:`view`.

After :ref:`reconciliation <accounting/payments/reconciling-payments>`, the invoice/bill is marked

> as :guilabel:`Paid`.

Reviewer: I would keep the reconciliation information as it was previously (in the tabs) because the two types of reconciliation are different, and it's useful for the user to know exactly what is reconciled with what.
```suggestion
```

      .. tip::
         If the main bank account is set as the outstanding account on the bank journal's payment
         method, registering the full payment on an invoice or bill moves the invoice/bill directly

>          to the :guilabel:`Paid` status without requiring bank reconciliation.

Reviewer: I would keep this tip under the "Using outstanding accounts" tab because it applies only when outstanding accounts are used.

      .. note::
         - If a bank transaction is reconciled in a different currency, a journal entry is
           automatically created to post the currency exchange gains/loss amount.
         - When a bank transaction is reconciled with an invoice with cash-basis, a journal entry is
           automatically created to post the cash-basis tax amount.
         - Unreconciling a payment unlinks it from the invoice or bill but does not delete the
           payment.
         - If a payment is (un)reconciled in a different currency, a journal entry is automatically
           created to post the currency exchange gains/losses (reversal) amount.
         - If a payment is (un)reconciled on an invoice with cash-basis taxes, a journal entry is

>            automatically created to post the cash-basis tax (reversal) amount.

Reviewer: These notes should be kept under the "Without outstanding accounts" or the "Using outstanding accounts" because they apply only to those cases.
The first two apply to reconciling with the bank transaction, and the last 3 apply when reconciling/unreconciling the invoice with the payment.

.. _accounting/payments/online:

Online payments
---------------

To make it more convenient for your customers to pay the invoices you issue, you can activate the
:guilabel:`Invoice Online Payment` feature, which adds a :guilabel:`Pay Now` button on their
customer portal. This allows your customers to see their invoices online and pay directly with

> their favorite payment method, making the payment process much easier.

Reviewer: We tend to avoid using "you/your" when it's possible.
```suggestion
To make invoice payment more convenient, the :guilabel:`Invoice Online Payment` feature can be
enabled, adding a :guilabel:`Pay Now` button to the customer portal. This allows customers to view 
invoices online and pay directly using their preferred payment method, simplifying the payment 
process.
```

Make sure your :ref:`payment providers <payment_providers/supported_providers>` are

> correctly configured.

Reviewer: same comment about you/your.
```suggestion
Make sure :ref:`payment providers <payment_providers/supported_providers>` are configured correctly.
```

> To activate the online payment, go to :menuselection:`Accounting --> Configuration --> Settings`,
> and scroll down to the :guilabel:`Customer Payments` section to enable
> :guilabel:`Invoice Online Payment`, and click :guilabel:`Save`.

Reviewer: ```suggestion
To activate the online payment, go to :menuselection:`Accounting --> Configuration --> Settings`,
scroll down to the :guilabel:`Customer Payments` section to enable :guilabel:`Invoice Online 
Payment`, and click :guilabel:`Save`.
```

After creating the invoice, click :guilabel:`Send` to send it to the customer by email.
They receive an email with a link that redirects them to the invoice on their customer
portal.

They can choose which :ref:`Payment Provider <payment_providers/online_providers>` to use by

> clicking :guilabel:`Pay Now`.

Reviewer: To simplify a little bit.
```suggestion
After creating the invoice, click :guilabel:`Send` to email it to the customer, who receives an
email with a link to view the invoice on their customer portal. From there, they can click 
:guilabel:`Pay Now` to select a :ref:`Payment Provider <payment_providers/online_providers>`.
```

.. note::
   By default, :ref:`Wire Transfer <payment_providers/bank_payments/wire_transfer>` is the

>    only payment provider activated, but you still have to fill out the payment details.

Reviewer: same comment about you/your.
```suggestion
   By default, :ref:`Wire Transfer <payment_providers/bank_payments/wire_transfer>` is the only enabled
   payment provider, but payment details must still be entered.
```

.. seealso::
   :doc:`/applications/finance/payment_providers`

As soon as the customer has paid, a payment for the invoice is registered and the invoice is marked

> as :guilabel:`In payment`.

Reviewer: ```suggestion
As soon as the customer has paid, a payment for the invoice is registered, and the invoice is marked
as :guilabel:`In payment`.
```

.. tabs::

   .. group-tab:: Without outstanding accounts

>       If the payment method configured in the payment provider has no outstanding accounts:

Reviewer: ```suggestion
      If the payment method configured in the payment provider has no :ref:`outstanding accounts
      <accounting/journals/outstanding-accounts>`:
```

      - No journal entry created.
      - The :guilabel:`Amount Due` is not updated.

   .. group-tab:: Using outstanding accounts

>       If the payment method configured in the payment provider has an outstanding account:

Reviewer: ```suggestion
      If the payment method configured in the payment provider has :ref:`outstanding accounts
      <accounting/journals/outstanding-accounts>`:
```

      - A journal entry is created.
      - The counterpart is reflected in an
        :ref:`outstanding receipts or payments account <accounting/journals/outstanding-accounts>`.
      - The :guilabel:`Amount Due` is reduced based on the payment amount.

After :ref:`reconciliation <accounting/payments/reconciling-payments>`, the invoice is marked as

> :guilabel:`Paid`.

Reviewer: I would add this reconciliation information inside the tabs and specify what is reconciled with what, so the user has the exact information.
```suggestion
```

.. _accounting/payments/not-tied:

Registering payments not tied to an invoice or bill
===================================================

When a new payment is registered via :menuselection:`Accounting --> Customers --> Payments` or
:menuselection:`Accounting --> Vendor --> Payments`, it is not directly linked to an invoice or
bill.

.. tabs::

   .. group-tab:: Without outstanding accounts

      Stand-alone payments should not be registered without using
      :ref:`outstanding accounts <accounting/journals/outstanding-accounts>` as:

>       - No entries are created.

Reviewer: ```suggestion
      - No journal entries are created.
```

      - The payment cannot be linked to the invoice/bill.
      - The amount paid or received is not reflected in the accounting.

>       - :guilabel:`Amount Due` is not updated based on the payment amount.

Reviewer: ```suggestion
      - The :guilabel:`Amount Due` is not updated based on the payment amount.
```

   .. group-tab:: Using outstanding accounts

>       - A journal entry is created.

Reviewer: ```suggestion
      For stand-alone payments using :ref:`outstanding accounts 
      <accounting/journals/outstanding-accounts>`:
      
      - A journal entry is created.
```

      - The payment's journal entry matches the :guilabel:`outstanding account` with the
        account receivable or the account payable until the payment is manually matched with its

>         related invoice or bill.

Reviewer: ```suggestion
      - The payment's journal entry matches the :ref:`outstanding accounts 
        <accounting/journals/outstanding-accounts>` with the account receivable or the account 
        payable until the payment is manually matched with its related invoice or bill.
```

After :ref:`reconciliation <accounting/payments/reconciling-payments>`, the invoice/bill is marked

> as :guilabel:`Paid`.

Reviewer: I would keep it as it was before, because there's no invoice or bill in this case. And the information applies only to the "Using outstanding accounts" tab.
```suggestion
```

   .. group-tab:: Using outstanding accounts

>       A blue banner appears on top of the invoice/bill when validating it and an

Reviewer: ```suggestion
      A blue banner appears on top of the invoice/bill when validating it, and an
```

      The :guilabel:`Payments matching` or :guilabel:`Auto-reconcile` tool allows reconciling
      journal items with each other (i.e., payments with customer invoices or vendor bills) either
      individually or in batches:

      #. Access the :guilabel:`Accounting Dashboard`.
      #. Click the :icon:`fa-ellipsis-v` (:guilabel:`ellipsis`) button from the :guilabel:`Customer
         Invoices` or :guilabel:`Vendor Bills` journals.

>       #. And select :guilabel:`Payments Matching`.

Reviewer: It would be best to keep steps 2 & 3 together, since this is the menu where "Payments Matching" is selected and should be a single action.
```suggestion
      #. Click the :icon:`fa-ellipsis-v` (:guilabel:`ellipsis`) button from the :guilabel:`Customer
         Invoices` or :guilabel:`Vendor Bills` journals, and select :guilabel:`Payments Matching`.
```

-----------------

The :guilabel:`Payments matching` tool opens all unreconciled journal items and allows them to be
processed individually, matching all payments and journal items. Go to
:menuselection:`Accounting --> Accounting --> Reconcile` or go to the
:guilabel:`Accounting Dashboard` click the :icon:`fa-ellipsis-v` (:guilabel:`ellipsis`) button from
the :guilabel:`Sales` or :guilabel:`Purchases` journals, and select

> :guilabel:`Payments Matching`.

Reviewer: ```suggestion
:guilabel:`Accounting Dashboard`, click the :icon:`fa-ellipsis-v` (:guilabel:`ellipsis`) button from
the :guilabel:`Sales` or :guilabel:`Purchases` journals, and select :guilabel:`Payments Matching`.
```

.. _accounting/payments/reconciling-payments:

Reconciling payments with bank transactions
===========================================

Once a payment has been registered, the status of the invoice or bill is :guilabel:`In payment`. The
next step is :doc:`reconciling <bank/reconciliation>` to finalize the payment workflow which will
mark the invoice or bill as :guilabel:`Paid`.

.. tabs::

   .. group-tab:: Without outstanding accounts

      - Create a bank transaction that matches the invoice information (amount due and partner).
        Once created, it is automatically reconciled with the invoice/bill marking it as
        :guilabel:`Paid`.
      - Or manually reconcile a related :doc:`bank transaction <bank/transactions>` line with the
        invoice or bill.

   .. group-tab:: Using outstanding accounts

      - Create a bank transaction that matches the payment information (amount due and partner).
        Once created, it is automatically reconciled with the payment marking the related
        invoice/bill as :guilabel:`Paid`.
      - Or manually reconcile the payment with a related :doc:`bank transaction <bank/transactions>`
        line.

.. seealso::

>  :doc:`Bank reconciliation <bank/reconciliation>`

Reviewer: I'd keep what was previously written, since we shouldn't describe how to create bank transactions here, as it's already documented in the Bank reconciliation page. And also because the difference was made on what had to be reconciled with what, whether there was an outstanding account or not.
```suggestion
.. tabs::

   .. group-tab:: Without outstanding accounts

      Once a payment has been registered, the status of the invoice or bill is :guilabel:`In
      payment`. The next step is :doc:`reconciling <bank/reconciliation>` the related :doc:`bank
      transaction <bank/transactions>` line with the invoice or bill to finalize the payment
      workflow and mark the invoice or bill as :guilabel:`Paid`.

   .. group-tab:: Using outstanding accounts

      Once a payment has been registered, the status of the invoice or bill is :guilabel:`In
      payment`. The next step is :doc:`reconciling <bank/reconciliation>` the payment with the
      related :doc:`bank transaction <bank/transactions>` line to finalize the payment workflow and
      mark the invoice or bill as :guilabel:`Paid`.
```
